src/ripple/app/data/DatabaseCon.h

```diff
@@ -85,27 +85,27 @@ class DatabaseCon
     };
 
     DatabaseCon (Setup const& setup,
-            std::string const& name,
-            const char* initString[],
-            int countInit);
+                 std::string const& name,
+                 const char* initString[],
+                 int countInit);
 
     soci::session& getSession()
     {
-        return *session_;
+        return session_;
     }
 
-    LockedSociSession checkoutDb()
+    LockedSociSession checkoutDb ()
     {
-        return LockedSociSession(session_.get (), lock_);
+        return LockedSociSession (&session_, lock_);
     }
 
     void setupCheckpointing (JobQueue*);
+
 private:
-    std::unique_ptr<WALCheckpointer> walCheckpointer_;
-    // shared_ptr to handle lifetime issues with walCheckpointer_
-    // never null.
-    std::shared_ptr<soci::session> session_;
     LockedSociSession::mutex lock_;
+
+    soci::session session_;
+    std::unique_ptr<Checkpointer> checkpointer_;
 };
```

Comment:
If we swap the declaration order of `session_` and `checkpointer_` then `checkpointer_` will hold a stale ref to `session_` when `DatabaseCon` is destroyed. I'm OK getting rid of the `shared_ptr` that prevented this, but we should add a comment that declaration order matters here.